apps/jobs/admin.py

```diff
@@ -62,39 +62,8 @@ def get_challenge_name_and_id(self, obj):
 
     def submit_job_to_worker(self, request, queryset):
         for submission in queryset:
-            challenge_id = submission.challenge_phase.challenge.id
-            challenge_phase_id = submission.challenge_phase.id
-            submission_id = submission.id
-            logger.info(
-                "[x] Received submission message with challenge id {}, challenge phase id {}, submission id {}".format(
-                    challenge_id, challenge_phase_id, submission_id
-                )
-            )
-            message = {
-                "challenge_pk": challenge_id,
-                "phase_pk": challenge_phase_id,
-                "submission_pk": submission.id,
-            }
-
-            if submission.challenge_phase.challenge.is_docker_based:
-                try:
-                    response = requests.get(submission.input_file.url)
-                except Exception as e:
-                    messages.error(
-                        request,
-                        "Failed to get input_file with exception: {0}".format(
-                            e
-                        ),
-                    )
-                    return
-
-                if response and response.status_code == 200:
-                    message["submitted_image_uri"] = response.json()[
-                        "submitted_image_uri"
-                    ]
-
+            message = handle_submission_rerun(submission)
             publish_submission_message(message)
-            queryset.update(status=Submission.SUBMITTED)
 
     submit_job_to_worker.short_description = "Run selected submissions"
```

Comment:
Please change the text to `Run selected submissions (will set the status to canceled for existing submissions)`